src/Optimizer/Transformer/GoogleFontsPreconnect.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace AmpProject\Optimizer\Transformer;
+
+use AmpProject\Dom\Document;
+use AmpProject\Optimizer\ErrorCollection;
+use AmpProject\Optimizer\Transformer;
+
+/**
+ * Transformer adding resource hints to preconnect to the Google Fonts domain.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class GoogleFontsPreconnect implements Transformer
+{
+
+    /**
+     * Domain that the Google Fonts static files are loaded from.
+     *
+     * @var string
+     */
+    const GOOGLE_FONTS_STATIC_DOMAIN = 'https://fonts.gstatic.com';
+
+    /**
+     * Domain that the Google Fonts API is accepting requests from.
+     *
+     * @var string
+     */
+    const GOOGLE_FONTS_API_DOMAIN = 'https://fonts.googleapis.com/';
```

Comment:
Doesn't work as well as above, so perhaps `BASE_URL`?

```suggestion
    const GOOGLE_FONTS_API_BASE_URL = 'https://fonts.googleapis.com/';
```